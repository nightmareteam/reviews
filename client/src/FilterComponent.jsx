import React from 'react';
//import styled from 'styled-components';
const styled = window.styled;
import Filter from './Filter.jsx';
import ActiveFilters from './ActiveFilters.jsx';
import Chart from './Chart.jsx';

/**
 * This component renders available and active filters.
 * @param {Function} setFilters - function to update state with the selected filters
 * @param {Integer} count - the review count given the selected filters
 * @param {Array} filters - an array of filter objects {id, displayName}
 * @param {Object} activeFilters - an object of active filters { id, options: {optionId, optionName}}
 */
const FilterComponent = ({setFilters, count, filters, activeFilters, sort}) => (
  <FilterContainer className='FilterContainer'>
    <Chart />
    <FilterList className='FilterList'>
      {
        filters.map(filter => (<Filter setFilters={setFilters} key={filter.id} filter={filter} />))
      }
      <SummaryType>
        {'DISPLAY AS:'}
        <SummaryList onChange={(e) => sort(e)}>
          <SummaryOption value="helpful" default>Most Helpful</SummaryOption>
          <SummaryOption value="recent" default>Recent</SummaryOption>
          <SummaryOption value="funny" default>Funny</SummaryOption>
        </SummaryList>
      </SummaryType>
    </FilterList>
    {
      Object.keys(activeFilters).length > 0 ? <ActiveFilters setFilters={setFilters} activeFilters={activeFilters} /> : null
    }
    <FilterSummary>
      {'Showing '}
      <FilterCount>{count}</FilterCount> 
      {' reviews that match the filters above'} 
    </FilterSummary>
    <Divider />
  </FilterContainer>
);
const FilterList = styled.div`
  background: #1f2e42;
  min-width: 526px;
`;

const FilterContainer = styled.div`
`;

const FilterSummary = styled.div`
  padding-top: 10px;
  padding-bottom: 10px;
  color: #c6d4df;
  font-size: 15px;
  display: block;
`;

const FilterCount = styled(FilterSummary)`
  font-weight: bold;
  display: inline-block;
`;

const SummaryType = styled.div`
  display: inline-block;
  position: relative;
  border-radius: 0px;
  display: inline-block;
  color: #4582a5;
  padding: 3px 25px 7px 10px;
  background: inherit;
  font-size: 10px;
  border-left: 1px solid #2a475e
`;

const SummaryList = styled.select`
  width: 100px;
  background: #4582a5;
  font-size: 12px;
  border: none;
  border-radius: 2px;
  margin-left: 5px;
  margin-top: 2px;
`;

const Divider = styled.div`
  margin-bottom: 0px;

  @media only screen and (min-width: 768px) {
    height: 1px;
    background: rgba( 0, 0, 0, 0.5 );
    margin-bottom: 20px;
  }
`;

const SummaryOption = styled.option``;
export default FilterComponent;