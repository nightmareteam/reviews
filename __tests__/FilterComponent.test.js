import React from 'react';
import { shallow } from 'enzyme';
import FilterComponent from '../client/src/FilterComponent.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props;
let mockCallback;

const createProps = (mockCallback) => ({
  setFilters: mockCallback,
  count: 0,
  filters: [
    {
      id: 'recommended',
      displayName: 'Review Type', 
      options: [
        {
          id: '1',
          displayName: 'Positive',
          count: 48
        },
        {
          id: '0',
          displayName: 'Negative',
          count: 52
        }
      ]
    },
    {
      id: 'language',
      displayName: 'Language', 
      options: [
        {
          'id': 'arabic',
          'displayName': 'Arabic',
          'count': 4
        },
        {
          'id': 'armenian',
          'displayName': 'Armenian',
          'count': 1
        }
      ]
    },
  ],
  activeFilters: {}
});

beforeEach(() => {
  jest.resetModules();
  mockCallback = jest.fn();
  props = createProps(mockCallback);
  wrapper = shallow(<FilterComponent {...props} /> );
});

describe('<FilterComponent /> rendering', () => {
  it('should render elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});