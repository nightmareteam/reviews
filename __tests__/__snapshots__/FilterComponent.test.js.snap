// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterComponent /> rendering should render elements 1`] = `
<styled.div
  className="FilterContainer"
>
  <Chart />
  <styled.div
    className="FilterList"
  >
    <Filter
      filter={
        Object {
          "displayName": "Review Type",
          "id": "recommended",
          "options": Array [
            Object {
              "count": 48,
              "displayName": "Positive",
              "id": "1",
            },
            Object {
              "count": 52,
              "displayName": "Negative",
              "id": "0",
            },
          ],
        }
      }
      key="recommended"
      setFilters={[MockFunction]}
    />
    <Filter
      filter={
        Object {
          "displayName": "Language",
          "id": "language",
          "options": Array [
            Object {
              "count": 4,
              "displayName": "Arabic",
              "id": "arabic",
            },
            Object {
              "count": 1,
              "displayName": "Armenian",
              "id": "armenian",
            },
          ],
        }
      }
      key="language"
      setFilters={[MockFunction]}
    />
    <styled.div>
      DISPLAY AS:
      <styled.select
        onChange={[Function]}
      >
        <styled.option
          default={true}
          value="helpful"
        >
          Most Helpful
        </styled.option>
        <styled.option
          default={true}
          value="recent"
        >
          Recent
        </styled.option>
        <styled.option
          default={true}
          value="funny"
        >
          Funny
        </styled.option>
      </styled.select>
    </styled.div>
  </styled.div>
  <styled.div>
    Showing 
    <Styled(styled.div)>
      0
    </Styled(styled.div)>
     reviews that match the filters above
  </styled.div>
  <styled.div />
</styled.div>
`;
